gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -1039,13 +1034,33 @@ def calculate_update_time_ranges(
 
     # CHECKS
 
+    def parsing_quality_check(self,dataset: xr.Dataset):
+        """
+        Master function containing quality checks general to all datasets
+        If successful passes without comment. If unsuccessful raises a descriptive error message.
+
+        Parameters
+        ----------
+        original_dataset : xr.Dataset
+            The final dataset to be parsed
+        """
+        # Aggressively assert that the data is in contiguous time order. Only valid if the dataset's time dimension is longer than 1 element
+        # This is to protect against data corruption, especially during insert and append operations, but it should probably be replaced with 
+        # a more sophisticated set of flags that let the user decide how to handle time data at their own risk.
+        times = dataset[self.time_dim].values
+        if len(times) >= 2:
+            # Check is only valid if we have 2 or more values with which to calculate the delta
+            expected_delta = times[1] - times[0]
+            if not self.are_times_in_expected_order(times=times, expected_delta=expected_delta):
```

Comment:
I think we should be branching here rather than passing the unused expected_delta value